internal/channel/channel.go

```diff
@@ -13,38 +19,88 @@ type Channel struct {
 	Type   string `db:"type"`
 	Config string `db:"config" json:"-"` // excluded from JSON config dump as this may contain sensitive information
 
-	plugin Plugin
+	Logger *logging.Logger
+	cmd    *exec.Cmd
+	rpc    *rpc.RPC
+	mu     sync.Mutex
 }
 
-func (c *Channel) GetPlugin() (Plugin, error) {
-	if c.plugin == nil {
-		f := Channels[c.Type]
-		if f == nil {
-			return nil, fmt.Errorf("unknown channel type %q", c.Type)
-		}
+func (c *Channel) StartPlugin(pluginDir string) error {
+	if c.cmd != nil {
+		return nil
+	}
 
-		p, err := f(c.Config)
-		if err != nil {
-			return nil, fmt.Errorf("error initializing channel type %q: %w", c.Type, err)
-		}
+	path := filepath.Join(pluginDir, c.Type)
+	logger := c.Logger.With(zap.String("cmd", path))
+
+	cmd := exec.Command(path)
+
+	writer, err := cmd.StdinPipe()
+	if err != nil {
+		return fmt.Errorf("failed to create stdin pipe: %w", err)
+	}
+
+	reader, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("failed to create stdout pipe: %w", err)
+	}
+
+	errPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("failed to create stderr pipe: %w", err)
+	}
+
+	go forwardLogs(errPipe, logger)
 
-		c.plugin = p
+	if err = cmd.Start(); err != nil {
+		return fmt.Errorf("failed to start cmd: %w", err)
 	}
+	logger.Debug("Cmd started successfully")
 
-	return c.plugin, nil
+	c.mu.Lock()
+	c.cmd = cmd
+	c.rpc = rpc.NewRPC(writer, reader, logger)
+	c.mu.Unlock()
+
+	if err = c.SetConfig(c.Config); err != nil {
+		c.ResetPlugin()
+
+		return fmt.Errorf("failed to set config: %w", err)
+	}
+	logger.Debug("Successfully set config: ", c.Config)
+
+	return nil
 }
 
 func (c *Channel) ResetPlugin() {
-	c.plugin = nil
-}
+	if c.cmd != nil {
```

Comment:
Same here, it should also lock `c.mu` before accessing `c.cmd`. (Hint: you have to be careful with the locking when calling this function from `StartPlugin()`.)